### PR TITLE
Test should not fail because it didn't match everything in the whitelist

### PR DESCRIPTION
**Pull request by qooplmao**
| Q             | A
| ------------- | ---
| Branch        | master for features and deprecations
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Documented?   | yes
| Fixed tickets | #866

`git_branch_name` fails if any of the items in the list fail.

PR changes so that any matching item in the whitelist will return a success unless already featured in the blacklist.

### CODE REVIEW

File: src/Task/Git/BranchName.php [hunk 1]
```diff
@@ -95,17 +96,24 @@ public function run(ContextInterface $context): TaskResultInterface
...
             $regex = new Regex($rule);
 
             $additionalModifiersArray = array_filter(str_split($config['additional_modifiers']));
             array_map([$regex, 'addPatternModifier'], $additionalModifiersArray);
-
-            if (!preg_match((string) $regex, $name)) {
-                $errors[] = sprintf('Whitelist rule not matched: %s', $rule);
+            
+            if (preg_match((string) $regex, $name)) {
+                if ($isBlacklisted) {
```
Comment (veewee) [inline src/Task/Git/BranchName.php:109]: Can you add some tests that show what happens if there is a mix of whitelist and blacklist and how it resolves a conflict in between these regexes?

**Review by veewee — COMMENTED** [review]
It indeed makes more sense to result OK on *any* whitelist item.
Thanks!

File: src/Task/Git/BranchName.php (hunk 1 above)
Comment (qooplmao) [reply src/Task/Git/BranchName.php:109]: I've added a test with a conflict and additional output.  
Does that cover what you are thinking?

**Review by Landerstraeten — APPROVED** [review]

**Comment by veewee** [comment]
Thanks!

**Pull request merged**